core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -835,8 +835,15 @@ private Optional<Expression> coerceComparisonWithRounding(
             }
             Type valueType = nullableValue.getType();
             Object value = nullableValue.getValue();
-            return floorValue(valueType, symbolExpressionType, value)
-                    .map(floorValue -> rewriteComparisonExpression(symbolExpressionType, symbolExpression, valueType, value, floorValue, comparisonOperator));
+            Optional<Object> floorValueOptional;
+            try {
+                floorValueOptional = floorValue(valueType, symbolExpressionType, value);
+            }
+            catch (TrinoException ignored) {
+                // Exception here means there's no such value at symbolExpressionType
+                return Optional.of(FALSE_LITERAL);
```

Comment:
Should this be only for INVALID_CAST_ARGUMENT or something like that?

Reply:
I agree, done